SWATCH-3338: Add billing_account_id to SkuCapacity in subscriptions api v2

Sgitario opened the pull request:
Jira issue: SWATCH-3338

## Description
When querying subscriptions for a given product id, the billing_account_id should be included in the result. This will enable filtering the subscription list on the GUI side without separate calls every time the billing_account_id is changed.

## Testing
IQE Test MR: https://gitlab.cee.redhat.com/insights-qe/iqe-rhsm-subscriptions-plugin/-/merge_requests/1060

## Manual

1.- Make sure you have a subscription in your database
2.- Start API service or swatch-contracts service
3.- Opt In for org ID of the subscription present in database
3.- Call the subscription api v2 endpoint:

```
curl -X GET \
  'http://localhost:8101/api/rhsm-subscriptions/v1/subscriptions/products/rosa' \
  -H 'accept: application/vnd.api+json' \
  -H 'x-rh-identity: <IDENTITY>'
```

And the reponse should include the new billing_account_id field in the capacity.

InsightsDroid commented:
### IQE Tests Summary Report
 9 failed, 182 passed, 2 skipped, 241 deselected, 4094 warnings in 3698.63s (1:01:38) 
Result: Failed
IQE plugin image: https://quay.io/cloudservices/iqe-tests:rhsm-subscriptions
|Failures|
|:-| 
|integration/exports/test_export_subscriptions.py::test_export_subscriptions_payg_filters[rhods]| 
|integration/swatch_contracts/test_contracts.py::test_verify_contract_no_metrics[aws-rosa]| 
|integration/swatch_contracts/test_contracts.py::test_verify_subscriptions_sync_for_contracts[rosa]| 
|integration/swatch_contracts/test_update_contract.py::test_verify_update_offering_description_will_modify_sub_table_product_name| 
|integration/swatch_contracts/test_update_contract.py::test_verify_contract_multiple_updates[aws-ansible-aap-managed]| 
|integration/swatch_tally/test_tally_ansible.py::test_verify_ansible_payg_usage| 
|integration/swatch_tally/test_tally_rhel_payg.py::test_verify_rhel_payg_via_cost_mgmt[aws-rhel-for-x86-els-payg]| 
|integration/swatch_tally/test_tally_rhel_payg.py::test_verify_rhel_payg_via_cost_mgmt[aws-rhel-for-x86-els-payg-addon]|

Sgitario commented:
/retest

InsightsDroid commented:
### IQE Tests Summary Report
 1 failed, 191 passed, 2 skipped, 241 deselected, 4151 warnings in 3700.85s (1:01:40) 
Result: Failed
IQE plugin image: https://quay.io/cloudservices/iqe-tests:rhsm-subscriptions-pr-4218
|Failures|
|:-| 
|integration/swatch_tally/test_tally_dedup_rhel_els.py::test_verify_deduplication_for_same_instance_different_event_source_without_any_existing_data|

InsightsDroid commented:
### IQE Tests Summary Report
 1 failed, 190 passed, 3 skipped, 241 deselected, 4151 warnings in 3482.46s (0:58:02) 
Result: Failed
IQE plugin image: https://quay.io/cloudservices/iqe-tests:rhsm-subscriptions-pr-4218
|Failures|
|:-| 
|component/swatch_contracts/test_offering.py::test_rhel_unlimited_sku_capacity|

liwalker-rh commented:
/retest

InsightsDroid commented:
### IQE Tests Summary Report
 192 passed, 2 skipped, 241 deselected, 4151 warnings in 3467.35s (0:57:47) 
Result: Success
IQE plugin image: https://quay.io/cloudservices/iqe-tests:rhsm-subscriptions-pr-4218

Sgitario commented:
> I think we also need to have the same change below so that when we switch to swatch-contracts there are no broken functionalities.
> 
> https://github.com/RedHatInsights/rhsm-subscriptions/blob/169b8b0d6cd935e0549be40bf37205650a0a70ba/swatch-contracts/src/main/java/com/redhat/swatch/contract/resource/api/v2/SubscriptionTableControllerV2.java#L277
> 

Oops... That's why copying logic instead of moving it is always a bad choice... Good catch.

> A minor note: While this change seems quite intuitive and might not require detailed test steps, however including them can be valuable for future reference. When you have a chance, please add api request and expected output. I believe we still add those to the PR even if we have IQE test. :)

Yeah, I used to write some PR instructions, but I thought this was simple enough. I've added some.

InsightsDroid commented:
### IQE Tests Summary Report
 1 failed, 191 passed, 2 skipped, 241 deselected, 4152 warnings in 3631.77s (1:00:31) 
Result: Failed
IQE plugin image: https://quay.io/cloudservices/iqe-tests:rhsm-subscriptions-pr-4218
|Failures|
|:-| 
|component/swatch_billable_usage/test_swatch_billable_usage.py::test_tally_summary_null_message|

InsightsDroid commented:
### IQE Tests Summary Report
 5 failed, 187 passed, 2 skipped, 241 deselected, 4151 warnings in 2839.32s (0:47:19) 
Result: Failed
IQE plugin image: https://quay.io/cloudservices/iqe-tests:rhsm-subscriptions-pr-4218
|Failures|
|:-| 
|integration/exports/test_export_subscriptions.py::test_export_subscriptions_filters[json-sku_filters0]| 
|integration/exports/test_export_subscriptions.py::test_export_subscriptions_filters[json-sku_filters1]| 
|integration/exports/test_export_subscriptions.py::test_export_subscriptions_filters[csv-sku_filters0]| 
|integration/exports/test_export_subscriptions.py::test_export_subscriptions_filters[csv-sku_filters1]| 
|integration/swatch_tally/test_tally_database.py::test_capacity_when_sku_updates|

InsightsDroid commented:
### IQE Tests Summary Report
 192 passed, 2 skipped, 241 deselected, 4151 warnings in 3763.21s (1:02:43) 
Result: Success
IQE plugin image: https://quay.io/cloudservices/iqe-tests:rhsm-subscriptions-pr-4218

liwalker-rh reviewed (approved):
Changes look good from QE side of things. Related IQE test is also ready to merge once this is merged.

Sgitario commented:
> Changes look good from QE side of things. Related IQE test is also ready to merge once this is merged.

Thanks @liwalker-rh ! 
I already addressed the feedback from @kartikshahc , so I'm merging these changes.

InsightsDroid commented:
### IQE Tests Summary Report
 1 failed, 194 passed, 2 skipped, 241 deselected, 4169 warnings in 3544.73s (0:59:04) 
Result: Failed
IQE plugin image: https://quay.io/cloudservices/iqe-tests:rhsm-subscriptions
|Failures|
|:-| 
|integration/swatch_conduit/test_ephemeral_conduit.py::test_e2e_conduit_ephermal|
